android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUITimePicker.java

```diff
@@ -123,16 +128,25 @@ public void processProperties(KrollDict d) {
                 maxDate = null;
             }   
         }
+        
+		Timer t = new Timer();
+		int delay = 0, rate = 500;
+		t.scheduleAtFixedRate(new TimerTask() {
+			@Override
+			public void run() {
+				detectTimeChange();
+			}
+		}, delay, rate);
```

Comment:
Don't use a Java "Timer". It executes the TimerTask on a separate thread, which will cause race conditions in our code. Use the Android "Handler" class instead which runs on the main UI thread (if created on the main thread). You'll want to use its postDelayed() method, but you'll have to call postDelayed() within the Runnable so that it'll schedule itself again.

Also, your current timer will run forever.  It'll continue to run after the view has been removed (this is bad).  You should start posting with the handler when the View.onAttachedToWindow() method gets called and stop the handler when the View.onDetachedFromWindow() method gets called.
https://developer.android.com/reference/android/view/View.html#onAttachedToWindow()
https://developer.android.com/reference/android/view/View.html#onDetachedFromWindow()